docker-compose.full-demo.yml

```diff
@@ -0,0 +1,158 @@
+    # So it needs to start once those are created.
+    depends_on:
+      - messenger
+      - notifier
+    volumes:
+      - './load_balancer/nginx.conf:/etc/nginx/nginx.conf:ro'
+
+    # The load balancer is the only service that has ports exposed out.
+    ports:
+      - '80:80'
+    environment:
+      - NGINX_PORT=80
+    networks:
+      - microservices_demo_1
+
+  rabbitmq:
+    image: 'rabbitmq:3.11.4-management-alpine'
+    container_name: rabbitmq
+    hostname: microservices_march
+
+    # The healthcheck is critical, because we need the RabbitMQ
+    # server to actually be up and accepting connections before we
+    # start the applications, otherwise they will crash.
+    healthcheck:
+      test: rabbitmq-diagnostics -q ping
+      interval: 30s
+      timeout: 30s
+      retries: 3
+    ports:
+      - '5672:5672'
+      - '15672:15672'
+    volumes:
+      - 'rabbit-data:/var/lib/rabbitmq/'
+      - 'rabbit-log:/var/log/rabbitmq/'
+    networks:
+      - microservices_demo_1
+
+  messenger:
+    image: microservices_demo/messenger
+    container_name: messenger
+
+    # This uses the `Dockerfile` in the `messenger` repository
+    # to build the container for the service.
+    # If you've made changes to the `messenger` application,
+    # run `docker-compose -f docker-compose.full-demo.yml build`
+    # to make sure this file is using the updated container.
+    build:
+      context: ../messenger
+    depends_on:
+      rabbitmq:
+        condition: service_healthy
+      notifier-db:
+        condition: service_healthy
+    environment:
+      PGPASSWORD: postgres
+      CREATE_DB_NAME: messenger
+      PGHOST: messenger-db
+      PGPORT: 5432
+      AMQPHOST: rabbitmq
+      AMQPPORT: 5672
+      PORT: 4000
+    networks:
+      - microservices_demo_1
+
+  notifier:
+    image: microservices_demo/notifier
+    container_name: notifier
+    build:
+      context: ../notifier
+    depends_on:
+      rabbitmq:
+        condition: service_healthy
+      notifier-db:
+        condition: service_healthy
+    environment:
+      PGPASSWORD: postgres
+      CREATE_DB_NAME: notifier
+      PGHOST: notifier-db
+      # Note that we are running this database on a nonstandard port to avoid conflicts
+      PGPORT: 5433
+      AMQPHOST: rabbitmq
+      AMQPPORT: 5672
+      PORT: 5000
+    networks:
+      - microservices_demo_1
+
+  messenger-db:
+    image: 'postgres:15'
+    container_name: messenger-db
+    healthcheck:
+      test: ["CMD-SHELL", "pg_isready -U postgres"]
+      interval: 30s
+      timeout: 30s
+      retries: 3
+    restart: always
+    environment:
+      POSTGRES_USER: postgres
+      POSTGRES_PASSWORD: postgres
+      POSTGRES_DB: messenger
+      PGDATA: /var/lib/postgresql/data/pgdata
+    ports:
+      - '5432:5432'
+    volumes:
+      - 'messenger-db-data:/var/lib/postgresql/data/pgdata'
+    networks:
+      - microservices_demo_1
+
+  notifier-db:
```

Comment:
Does this mean there will be two postgres containers running?

Reply:
Yes. With potgres in theory one could use a single container and have different databases - but I'm trying to show what I've experienced (although in a cloud setting) where individual apps own infrastructure that is specific to them.  That way one application hitting the DB hard wouldn't have the potential of causing issues for another.

Reply:
fair point, I didn't consider the microservice siloing